clerk/keeper.go

```diff
@@ -119,6 +119,32 @@ func (k *Keeper) GetAllEventRecords(ctx sdk.Context) (records []*types.EventReco
 	return
 }
 
+// GetEventRecordList returns all records with params like page and limit
+func (k *Keeper) GetEventRecordList(ctx sdk.Context, page uint64, limit uint64) ([]types.EventRecord, error) {
```

Comment:
Would be better I think if the keeper didn't need page and limit params. Can we add this logic in the querier instead of keeper?

Reply:
Storage should read only needed data. We shouldn't fetch all data from storage and filter at querier layer.

Reply:
Agreed.